### PR TITLE
{lib}[GCCcore/13.3.0] assimp v5.4.3, double-conversion v3.3.0, NSPR v4.35, ...

### PR DESCRIPTION
**Pull request by maxim-masterov**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by maxim-masterov** [comment]
Test report by @maxim-masterov
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
tcn101.local.snellius.surf.nl - Linux RHEL 9.4, x86_64, AMD EPYC 7H12 64-Core Processor, Python 3.9.18
See https://gist.github.com/maxim-masterov/195e1f9c1513f6d2c267822becf01d50 for a full test report.

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 7 out of 7 (7 easyconfigs in total)
zen2-rockylinux-810 - Linux Rocky Linux 8.10, x86_64, AMD EPYC 7452 32-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/SebastianAchilles/389bb9cd45174948718daddf89e36a7f for a full test report.

**Comment by SebastianAchilles** [comment]
@boegelbot please test @ jsc-zen3
CORE_CNT=16

**Comment by SebastianAchilles** [comment]
@boegelbot please test @ generoso

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 0 out of 0 (7 easyconfigs in total)
bwd-rockylinux-94 - Linux Rocky Linux 9.4 (Blue Onyx), x86_64, Intel(R) Core(TM) i7-6900K CPU @ 3.20GHz (broadwell), 2 x NVIDIA NVIDIA GeForce GTX 1060 6GB, 535.183.06, Python 3.9.18
See https://gist.github.com/SebastianAchilles/63c6d0623472209411c5aacabfc98b1c for a full test report.

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 0 out of 0 (7 easyconfigs in total)
skl-rockylinux-810 - Linux Rocky Linux 8.10, x86_64, Intel(R) Core(TM) i7-10700 CPU @ 2.90GHz (skylake), 1 x NVIDIA NVIDIA RTX A4000, 555.42.06, Python 3.6.8
See https://gist.github.com/SebastianAchilles/a9abc4df6a1069dc43c3daf54a52abfe for a full test report.

**Comment by SebastianAchilles** [comment]
> Test report by @maxim-masterov **FAILED** Build succeeded for 0 out of 1 (1 easyconfigs in total) tcn101.local.snellius.surf.nl - Linux RHEL 9.4, x86_64, AMD EPYC 7H12 64-Core Processor, Python 3.9.18 See https://gist.github.com/maxim-masterov/195e1f9c1513f6d2c267822becf01d50 for a full test report.

This test report failed with missing dependencies:
```
Missing modules for dependencies (use --robot?): re2c/3.1-GCCcore-13.3.0, double-conversion/3.3.0-GCCcore-13.3.0, assimp/5.4.3-GCCcore-13.3.0, NSS/3.104-GCCcore-13.3.0, snappy/1.2.1-GCCcore-13.3.0 (at easybuild/tools/toolchain/toolchain.py:563 in _check_dependencies)
```

**Comment by maxim-masterov** [comment]
@SebastianAchilles Yes, it was my bad. I didn't have time to re-generate the report after that. It takes almost 9 hours to build Qt6 on our system :|

**Comment by SebastianAchilles** [comment]
> @SebastianAchilles Yes, it was my bad. I didn't have time to re-generate the report after that. It takes almost 9 hours to build Qt6 on our system :| Please ignore that report, if you think it is redundant.

That is no problem. I just wanted to mention this, in case someone is wondering, why this test reported failed. I think we have enough test reports from different systems.

File: easybuild/easyconfigs/a/assimp/assimp-5.4.3-GCCcore-13.3.0.eb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    ('CMake', '3.29.3'),
+    ('zlib', '1.3.1'),
+]
+
+# workaround bug with GCC13 https://github.com/assimp/assimp/issues/5315
+configopts = "-DASSIMP_WARNINGS_AS_ERRORS=OFF "
+
+
+sanity_check_paths = {
+    'files': ['include/%(name)s/types.h', 'lib/libassimp.so'],
```
Comment (SebastianAchilles) [inline easybuild/easyconfigs/a/assimp/assimp-5.4.3-GCCcore-13.3.0.eb:32]:
```suggestion
    'files': ['include/%(name)s/types.h', 'lib/libassimp.%s' % SHLIB_EXT],
```

File: easybuild/easyconfigs/d/double-conversion/double-conversion-3.3.0-GCCcore-13.3.0.eb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    "-DCMAKE_POSITION_INDEPENDENT_CODE=ON -DCMAKE_STATIC_LIBRARY_SUFFIX_CXX=_pic.a",
+    '-DBUILD_SHARED_LIBS=ON',
+]
+
+
+sanity_check_paths = {
+    'files': ['include/%(name)s/bignum.h', 'include/%(name)s/cached-powers.h', 'include/%(name)s/diy-fp.h',
+              'include/%(name)s/%(name)s.h', 'include/%(name)s/fast-dtoa.h', 'include/%(name)s/fixed-dtoa.h',
+              'include/%(name)s/ieee.h', 'include/%(name)s/strtod.h', 'include/%(name)s/utils.h',
+              'lib/libdouble-conversion.a', 'lib/libdouble-conversion.so', 'lib/libdouble-conversion_pic.a'],
```
Comment (SebastianAchilles) [inline easybuild/easyconfigs/d/double-conversion/double-conversion-3.3.0-GCCcore-13.3.0.eb:33]:
```suggestion
    'files': ['include/double-conversion/%s.h' % h for h in ['bignum', 'cached-powers', 'diy-fp', 'double-conversion',
                                                             'fast-dtoa', 'fixed-dtoa', 'ieee', 'strtod', 'utils']] +
             ['lib/libdouble-conversion.%s' % e for e in ['a', SHLIB_EXT]] + ['lib/libdouble-conversion_pic.a'],
```

File: easybuild/easyconfigs/n/NSPR/NSPR-4.35-GCCcore-13.3.0.eb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    ('binutils', '2.42'),
+]
+
+configopts = "--disable-debug --enable-optimize --enable-64bit"
+
+
+sanity_check_paths = {
+    'files': ['bin/%(namelower)s-config', 'lib/libnspr%(version_major)s.a', 'lib/libnspr%(version_major)s.so',
+              'lib/libplc%(version_major)s.a', 'lib/libplc%(version_major)s.so', 'lib/libplds%(version_major)s.a',
+              'lib/libplds%(version_major)s.so', 'lib/pkgconfig/%(namelower)s.pc'],
```
Comment (SebastianAchilles) [inline easybuild/easyconfigs/n/NSPR/NSPR-4.35-GCCcore-13.3.0.eb:26]:
```suggestion
    'files': ['bin/nspr-config', 'lib/libnspr%(version_major)s.a', 'lib/libnspr%%(version_major)s.%s' % SHLIB_EXT,
              'lib/libplc%(version_major)s.a', 'lib/libplc%%(version_major)s.%s' % SHLIB_EXT,
              'lib/libplds%(version_major)s.a', 'lib/libplds%%(version_major)s.%s' % SHLIB_EXT,
              'lib/pkgconfig/%(namelower)s.pc'],
```

File: easybuild/easyconfigs/q/Qt6/Qt6-6.7.2-GCCcore-13.3.0.eb [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+configopts = "-Wno-dev -DFEATURE_qtpdf_build=OFF -DQT_AVOID_CMAKE_ARCHIVING_API=ON "
+configopts += "-DPython3_ROOT_DIR=$EBROOTPYTHON -DBUILD_qtwayland=OFF "
+
+
+# Removed from Qt6.0.0 but may be added back in the future
+# configopts += '-DBUILD_qtgamepad=OFF '  # Does not work on CentOS 7
+
+sanity_check_paths = {
+    'files': ['bin/qmake6', 'lib/libQt6Core.so', 'lib/libQt6WebEngineCore.so'],
```
Comment (SebastianAchilles) [inline easybuild/easyconfigs/q/Qt6/Qt6-6.7.2-GCCcore-13.3.0.eb:84]:
```suggestion
    'files': ['bin/qmake6', 'lib/libQt6Core.%s' % SHLIB_EXT, 'lib/libQt6WebEngineCore.%s' % SHLIB_EXT],
```

File: easybuild/easyconfigs/s/snappy/snappy-1.2.1-GCCcore-13.3.0.eb [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+]
+
+configopts = [
+    "-DSNAPPY_BUILD_TESTS=OFF -DSNAPPY_BUILD_BENCHMARKS=OFF",
+    "-DBUILD_SHARED_LIBS=ON -DSNAPPY_BUILD_TESTS=OFF -DSNAPPY_BUILD_BENCHMARKS=OFF",
+]
+
+
+sanity_check_paths = {
+    'files': ['lib64/libsnappy.a', 'lib64/libsnappy.so', 'include/%(name)s.h'],
```
Comment (SebastianAchilles) [inline easybuild/easyconfigs/s/snappy/snappy-1.2.1-GCCcore-13.3.0.eb:34]:
```suggestion
    'files': ['lib64/libsnappy.a', 'lib64/libsnappy.%s' % SHLIB_EXT, 'include/snappy.h'],
```

**Review by SebastianAchilles — CHANGES_REQUESTED** [review]
Just a few minor comments. I think the template `SHLIB_EXT` was removed. I think I have seen happening this before when using `--insert-checksums`. I think we should keep the `SHLIB_EXT` templates in the sanity checks.

File: easybuild/easyconfigs/a/assimp/assimp-5.4.3-GCCcore-13.3.0.eb (hunk 1 above)
Comment (maxim-masterov) [reply easybuild/easyconfigs/a/assimp/assimp-5.4.3-GCCcore-13.3.0.eb:32]: thanks, indeed, it was changed by `--copy-ec` option :(

**Comment by SebastianAchilles** [comment]
snappy-1.1.10-GCCcore-13.3.0.eb was already added in https://github.com/easybuilders/easybuild-easyconfigs/pull/21660 I think it makes sense to use only on version for the GCCcore 13.3.0 toolchain. Do you think it is better to use 1.2.1 or would it also be okay to use 1.1.10?

**Comment by SebastianAchilles** [comment]
> @SebastianAchilles I agree that it's logical to use one version. v1.2.1 should have better performance than v1.1.10, so I would stay with that version in 2024a. Also, v1.1.10 is over a year old and is already included in the 2023a and 2023b toolchains. I'm not sure if it makes sense to keep that version in 2024a.

Thanks for the explanation. I agree that is makes sense to use v1.2.1 :+1:

**Review by SebastianAchilles — APPROVED** [review]
lgtm

**Comment by SebastianAchilles** [comment]
Going in, thanks @maxim-masterov!

**Pull request merged**